File: app/src/main/java/com/estasvegano/android/estasvegano/data/web/request/AddProducerRequest.java
package com.estasvegano.android.estasvegano.data.web.request;

import android.os.Parcelable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import auto.parcel.AutoParcel;

@AutoParcel
@JsonDeserialize(builder = AddProducerRequest.Builder.class)
public abstract class AddProducerRequest implements Parcelable {

    public static Builder builder() {
        return new AutoParcel_AddProducerRequest.Builder().ethical(null);
    }

    @JsonProperty("title")
    public abstract String title();


    @JsonProperty("ethical")
    @Nullable
    public abstract Boolean ethical();

    @AutoParcel.Builder
    public static abstract class Builder {

        @JsonProperty("title")
        public abstract Builder title(@NonNull String title);

        @JsonProperty("ethical")
        public abstract Builder ethical(@Nullable Boolean ethical);

        @NonNull
        public abstract AddProducerRequest build();
    }
}

File: app/src/main/java/com/estasvegano/android/estasvegano/data/web/request/AddProducerRequest.kt
package com.estasvegano.android.estasvegano.data.web.request

import android.os.Parcel
import android.os.Parcelable

data class AddProducerRequest(val title: String, val ethical: Boolean? = null) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<AddProducerRequest> = object : Parcelable.Creator<AddProducerRequest> {
            override fun createFromParcel(source: Parcel): AddProducerRequest = AddProducerRequest(source)
            override fun newArray(size: Int): Array<AddProducerRequest?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readValue(Boolean::class.java.classLoader) as Boolean?
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeValue(ethical)
    }
}